### PR TITLE
lib/promscrape/discovery/kubernetes: fix watcher start order for roles endpoints and endpointslice

### PR DESCRIPTION
**Pull request by Haleygo**
follow up https://github.com/VictoriaMetrics/VictoriaMetrics/commit/632d788b63bcead6b1afa0ba8d4c2fe1a129ca92
Previously the groupWatcher for endpoints or endpointslice could be [mistakenly stopped](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/082892f20acdb6bb3fca70da117c35d05f7d33b4/lib/promscrape/discovery/kubernetes/api_watcher.go#L320-L344) 
when initial reloadObjects requests for pod or services resources take too long(more than 2*7s).

### CODE REVIEW

File: lib/storage/partition.go [hunk 1]
```diff
@@ -142,7 +142,6 @@ type partition struct {
 	partsLock sync.Mutex
 
 	// Contains inmemory parts with recently ingested data.
-	// It must be merged into either smallParts or bigParts to become visible to search.
```
Comment (Haleygo) [inline lib/storage/partition.go:145]: please see https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5410#issuecomment-1870971710

**Comment by hagen1778** [comment]
@f41gh7 @zekker6 I'd appreciate your review!

**Review by valyala — APPROVED** [review]

**Comment by valyala** [comment]
@Haleygo , thanks for the fix!

**Comment by valyala** [comment]
FYI, this pull request has been included in [v1.97.0](https://docs.victoriametrics.com/CHANGELOG.html).

**Comment by valyala** [comment]
This pull request has been also included in [v1.97.11 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.93.11).

**Pull request merged**